lib/devicetrust/native/device_darwin.h

```diff
@@ -0,0 +1,54 @@
+#ifndef DEVICE_DARWIN_H_
+#define DEVICE_DARWIN_H_
+
+#include <stddef.h>
+#include <stdint.h>
+
+// PublicKey represents a device public key, containing its credential ID and
+// binary representation (as returned by SecKeyCopyExternalRepresentation).
+// A device key is guaranteed to be stored in the Secure Enclave, and thus is an
+// ECDSA P-256 key.
+typedef struct PublicKey {
+  const char *id;
+  uint8_t *pub_key;
+  size_t pub_key_len;
+} PublicKey;
+
+struct _Bytes {
+  uint8_t *data;
+  size_t data_len;
+};
+
+// Digest is a hash digest to be signed.
+typedef struct _Bytes Digest;
+
+// Signature is a signature over a Digest.
+typedef struct _Bytes Signature;
+
+// DeviceKeyCreate creates a new Secure Enclave key for the device, or returns
+// the existing key if one is already present.
+// Callers are responsible for allocating and freeing all "out" parameters.
+// Returns zero for success, non-zero for failures (likely an OSStatus value).
+int64_t DeviceKeyCreate(const char *newID, PublicKey *pubKeyOut);
```

Comment:
nit: Do we expect over 4 billion errors? I'd just return `int`. For me `int64_t` looks like a function returns something different than just an error.

Reply:
Downgraded to int32_t - we can't go lower than that because of OSStatus. Good call.